File: src/main/java/io/tripled/social/client/application/DefaultFollowUserUseCase.java
package io.tripled.social.client.application;

import io.tripled.social.client.domain.SocialNetwork;
import io.tripled.social.client.domain.SocialNetworkRepository;
import io.tripled.social.client.domain.UserName;

public class DefaultFollowUserUseCase implements FollowUserUseCase {

  private SocialNetworkRepository socialNetworkRepository;

  public DefaultFollowUserUseCase(SocialNetworkRepository socialNetworkRepository) {
    this.socialNetworkRepository = socialNetworkRepository;
  }

  @Override
  public void follow(String userName, String userNameToFollow) {
    SocialNetwork socialNetwork = socialNetworkRepository.get();

    socialNetwork.follow(new UserName(userName), new UserName(userNameToFollow));
  }
}

File: src/main/kotlin/io/tripled/social/client/application/DefaultFollowUserUseCase.kt
package io.tripled.social.client.application

import io.tripled.social.client.domain.SocialNetwork
import io.tripled.social.client.domain.SocialNetworkRepository
import io.tripled.social.client.domain.UserName

class DefaultFollowUserUseCase(socialNetworkRepository: SocialNetworkRepository) : FollowUserUseCase {
    private val socialNetworkRepository: SocialNetworkRepository

    init {
        this.socialNetworkRepository = socialNetworkRepository
    }

    override fun follow(userName: String, userNameToFollow: String): String {
        val socialNetwork: SocialNetwork = socialNetworkRepository.get()

        return socialNetwork.follow(UserName(userName), UserName(userNameToFollow))
    }
}
